hicommonwealth/commonwealth
# Integrate api into myCommunityStake page

## Link to Issue
Closes: #7238

## Description of Changes
- Makes changes to GetStakeTransactions to pull the correct data for the front end
- Replaces the hardcoded sample data with the data returned from the API
- Modifies buildEtherscanLink to link to either blast or base depending on the chain node of the community

## Test Plan
Workflow:
- Create a community with stake and start buying/selling stake
- On transaction buy/sell check the etherscan page (pop-out icon). Make sure it navigates to the correct etherscan.
- Check this transaction page.

Manual:
- You can enter your own transactions into the database, just replace your address. For example:
```
insert into "StakeTransactions" values ('0x1234567891234567891234567891234567891234567891234567891234567891', 'zaks-community', 2, [Your address], 100, 1000, 'sell', 100);
insert into "StakeTransactions" values ('0x1234567891234567891234567891234567891234567891234567891234567893', 'jpegmullets', 2, [Your address], 100, 1000, 'buy', 100);
insert into "StakeTransactions" values ('0x1234567891234567891234567891234567891234567891224567891234567893', 'blast-off', 2, [Your address], 100, 1000, 'buy', 100);
```

- check that blast-off links to the blastscan page, and jpegmullets links to the basescan page by clicking on the right hand side of the transaction listing (window pop-out icon)

Looks like:
<img width="1340" alt="image" src="https://github.com/hicommonwealth/commonwealth/assets/14794654/5f0ee04c-c09a-4c0a-98d4-dc82f71fa96e">
<img width="1325" alt="image" src="https://github.com/hicommonwealth/commonwealth/assets/14794654/0f879c79-dfa9-429c-8fdc-242b93a04880">


## Deployment Plan
1. (Backfill transactions) We will need to go through base and blast and manually enter the transactions that were made before this PR had gone in
